design/2022_Q2_data_citations.md

```diff
@@ -0,0 +1,217 @@
+[notoriously difficult](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/)
+to represent - most humans have two names, but some have one, some have many, and legally a human
+can change their name to [almost anything](https://en.wikipedia.org/wiki/Prince_(musician)).
+As such, the standard convention of two name fields can't
+accommodate all human names. Furthermore, human names are not necessarily unique.
+
+One approach for representing names, as taken in the
+[KBase auth2 service](https://github.com/kbase/auth2), is to represent names by a single
+completely arbitrary string, which can handle most cases. We use the same approach here.
+
+Note that for all the cases below, the the EDUs in the `MongoDB` database are currently stored in
+a list embedded in a list of provenance actions, which means they are unindexable. No features in
+this proposal require indexing so this case is not addressed.
+
+1. Another option is making the identifier entries a structure instead of an arbitrary string.
+   Whether the identifier is resolvable by a network query could then be added as a field,
+   and the workspace could validate the identifier. This seems generally unnecessary, however -
+   it seems reasonable for the code saving the object to validate the identifier when necessary.
+
+### Author option 1 - arbitrary strings
+
+In this case the `authors` field is a list of arbitrary strings which will be populated
+verbatim from the data source.
+
+Pros:
+
+* Simple implementation
+
+Cons:
+
+* Names don't uniquely identify people
+* Names can change over time, leaving outdated records in the database
+* The same names can be represented differently, meaning the database could have different
+  records that refer to the same person
+* No possibility of adding new fields in the future
+  
+Estimated work: Conservatively 1 week
+
+### Author option 2 - structure
+
+In this case the `authors` field is a list of structures with the fields
+
+* `name` - an arbitrary string
+* `kbaseid` - the person's KBase ID, if any. Any entries will be validated as existing users.
+* `orcid` - the persons's OrcID, if any. See validations options below.
+
+Pros:
+
+* Relatively simple implementation
+* Allows for more information about the person, including theoretically unique IDs (although
+  a person can have many KBase accounts. A person *should* only have one OrcID but it's not clear
+  if that constraint is enforced or enforceable).
+* Allows for adding more fields in the future to add more information about people in data
+  citations.
+  
+Cons:
+
+* In many cases only the person's name will be available
+* Names don't uniquely identify people
+* Names can change over time, leaving outdated records in the database
+* The same names can be represented differently, meaning the database could have different
+  records that refer to the same person
+* The name could be entered differently for the same KBase or OrcID in different records
```

Comment:
how about adding a 'synonym' or 'aka'-type field, or having `name` as a list?

What is going to be done about merging possible duplicates or giving users the option of using an existing "author" object or creating a new one?

Reply:
> how about adding a 'synonym' or 'aka'-type field, or having name as a list?

Hmm, that could be done, especially an `additional_names` field or something like that.
That being said, I suspect this'll be an uncommon case and so personally I'd YAGNI it for the first iteration.

Reply:
> What is going to be done about merging possible duplicates or giving users the option of using an existing "author" object or creating a new one?

Nothing. Once the record is saved, it's saved and immutable. If a user wants to look at EDUs saved for other objects (either as provenance or post-save) they can, but there's not central record of users. That's option #3.